### PR TITLE
Intradoc links conversion

### LINKED ISSUE #1771: fix `cargo deadlinks` results
I suggest that when fixing, migrating to intra-doc links.

```
Found invalid urls in struct.NamedFile.html:
	Linked file at path :sync_all does not exist!
	Linked file at path /Users/rob/.cargo/target-shared/doc/io/index.html does not exist!
Found invalid urls in http/struct.HeaderMap.html:
	Linked file at path http/struct.GetAll.html does not exist!
Found invalid urls in error/index.html:
	Linked file at path error/Sender does not exist!
	Linked file at path error/Receiver does not exist!
Found invalid urls in error/struct.Canceled.html:
	Linked file at path error/Sender does not exist!
	Linked file at path error/Receiver does not exist!
Found invalid urls in web/index.html:
	Linked file at path web/Sender does not exist!
	Linked file at path web/Receiver does not exist!
Found invalid urls in trait.FromRequest.html:
	Linked file at path struct.PayloadConfig.html does not exist!
	Linked file at path struct.JsonConfig.html does not exist!
Found invalid urls in web/struct.Canceled.html:
	Linked file at path web/Receiver does not exist!
	Linked file at path web/Sender does not exist!
Found invalid urls in http/struct.HeaderMap.html:
	Linked file at path http/struct.GetAll.html does not exist!
Found invalid urls in dev/struct.JsonBody.html:
	Linked file at path dev/struct.JsonConfig.html does not exist!
Found invalid urls in error/index.html:
	Linked file at path error/Sender does not exist!
	Linked file at path error/Receiver does not exist!
Found invalid urls in dev/trait.Service.html:
	Linked file at path dev/fn.fn_service.html does not exist!
Found invalid urls in error/struct.Canceled.html:
	Linked file at path error/Sender does not exist!
	Linked file at path error/Receiver does not exist!
Found invalid urls in http/struct.HeaderMap.html:
	Linked file at path http/struct.GetAll.html does not exist!
```

### PR DESCRIPTION
**Pull request by realaravinth**
## PR Type
Intradoc links conversion and fixing broken links in documentation 

## PR Checklist
Check your PR fulfills the following:

- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt

## Overview

Intradoc links conversion is complete. 

`cargo deadlinks` deadlinks was run to check for broken links. All broken links are from the following reexports:

- `actix_service`(will fix this in a bit)
```
Found invalid urls in dev/trait.Service.html:
	Linked file at path dev/fn.fn_service.html does not exist!
```

- and a bunch of other reexports from `http`, `tokio`, `tokio-util`,
`indexmap`, `ppv_lite86`, `byteorder` and `futures_util`.

Closes #1771

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@
   <p>
 
 [![crates.io](https://img.shields.io/crates/v/actix-web?label=latest)](https://crates.io/crates/actix-web)
-[![Documentation](https://docs.rs/actix-web/badge.svg?version=3.3.2)](https://docs.rs/actix-web/3.3.2)
+[![Documentation](https://docs.rs/actix-web/badge.svg?version=3.3.2)](actix_web)
```
Comment (robjtede) [inline README.md]: The specific versions in the readme links are intentional. Please revert these readme changes as it's easier for me to update when releasing.

The reason is crates.io caches the readme and I want the docs link to reflect the crate version.

File: actix-http/src/header/map.rs [hunk 2]
```diff
@@ -9,7 +9,6 @@ use http::header::{HeaderName, HeaderValue};
 ///
 /// `HeaderMap` is an multi-map of [`HeaderName`] to values.
 ///
```
Comment (robjtede) [inline actix-http/src/header/map.rs:11]: Remove instances of training white space in doc comments.

File: actix-web-codegen/README.md [hunk 3]
```diff
@@ -3,7 +3,7 @@
 > Helper and convenience macros for Actix Web
 
 [![crates.io](https://meritbadge.herokuapp.com/actix-web-codegen)](https://crates.io/crates/actix-web-codegen)
-[![Documentation](https://docs.rs/actix-web-codegen/badge.svg)](https://docs.rs/actix-web)
+[![Documentation](https://docs.rs/actix-web-codegen/badge.svg)](https://docs.rs/actix-web-codegen/0.4.0/actix_web_codegen/)
```
Comment (realaravinth) [inline actix-web-codegen/README.md:6]: The original file was pointing to `actix_web`, if this was intentional I could revert it

Comment (robjtede) [reply actix-web-codegen/README.md:6]: no that was an oversight, thanks

**Comment by robjtede** [comment]
Getting there, I got some warnings when building for the first time:

```
› cargo doc --no-deps --open --workspace
   Compiling actix-web-codegen v0.4.0 (/Users/rob/Development/forks/actix-web/actix-web-codegen)
 Documenting actix-web-codegen v0.4.0 (/Users/rob/Development/forks/actix-web/actix-web-codegen)
    Checking actix-codec v0.3.0
    Checking actix-utils v2.0.0
warning: `get` is both a function and an attribute macro
  --> actix-web-codegen/src/lib.rs:49:12
   |
49 | //! [GET]: get
   |            ^^^ ambiguous link
   |
   = note: `#[warn(broken_intra_doc_links)]` on by default
help: to link to the function, add parentheses
   |
49 | //! [GET]: get()
   |            ^^^^^
help: to link to the attribute macro, prefix with `macro@`
   |
49 | //! [GET]: macro@get
   |            ^^^^^^^^^

warning: `post` is both a function and an attribute macro
  --> actix-web-codegen/src/lib.rs:50:13
   |
50 | //! [POST]: post
   |             ^^^^ ambiguous link
   |
help: to link to the function, add parentheses
   |
50 | //! [POST]: post()
   |             ^^^^^^
help: to link to the attribute macro, prefix with `macro@`
   |
50 | //! [POST]: macro@post
   |             ^^^^^^^^^^

warning: `patch` is both a function and an attribute macro
  --> actix-web-codegen/src/lib.rs:57:14
   |
57 | //! [PATCH]: patch
   |              ^^^^^ ambiguous link
   |
help: to link to the function, add parentheses
   |
57 | //! [PATCH]: patch()
   |              ^^^^^^^
help: to link to the attribute macro, prefix with `macro@`
   |
57 | //! [PATCH]: macro@patch
   |              ^^^^^^^^^^^

warning: `put` is both a function and an attribute macro
  --> actix-web-codegen/src/lib.rs:51:12
   |
51 | //! [PUT]: put
   |            ^^^ ambiguous link
   |
help: to link to the function, add parentheses
   |
51 | //! [PUT]: put()
   |            ^^^^^
help: to link to the attribute macro, prefix with `macro@`
   |
51 | //! [PUT]: macro@put
   |            ^^^^^^^^^

warning: `delete` is both a function and an attribute macro
  --> actix-web-codegen/src/lib.rs:52:15
   |
52 | //! [DELETE]: delete
   |               ^^^^^^ ambiguous link
   |
help: to link to the function, add parentheses
   |
52 | //! [DELETE]: delete()
   |               ^^^^^^^^
help: to link to the attribute macro, prefix with `macro@`
   |
52 | //! [DELETE]: macro@delete
   |               ^^^^^^^^^^^^

warning: `head` is both a function and an attribute macro
  --> actix-web-codegen/src/lib.rs:53:13
   |
53 | //! [HEAD]: head
   |             ^^^^ ambiguous link
   |
help: to link to the function, add parentheses
   |
53 | //! [HEAD]: head()
   |             ^^^^^^
help: to link to the attribute macro, prefix with `macro@`
   |
53 | //! [HEAD]: macro@head
   |             ^^^^^^^^^^

warning: `connect` is both a function and an attribute macro
  --> actix-web-codegen/src/lib.rs:54:16
   |
54 | //! [CONNECT]: connect
   |                ^^^^^^^ ambiguous link
   |
help: to link to the function, add parentheses
   |
54 | //! [CONNECT]: connect()
   |                ^^^^^^^^^
help: to link to the attribute macro, prefix with `macro@`
   |
54 | //! [CONNECT]: macro@connect
   |                ^^^^^^^^^^^^^

warning: `options` is both a function and an attribute macro
  --> actix-web-codegen/src/lib.rs:55:16
   |
55 | //! [OPTIONS]: options
   |                ^^^^^^^ ambiguous link
   |
help: to link to the function, add parentheses
   |
55 | //! [OPTIONS]: options()
   |                ^^^^^^^^^
help: to link to the attribute macro, prefix with `macro@`
   |
55 | //! [OPTIONS]: macro@options
   |                ^^^^^^^^^^^^^

warning: `trace` is both a function and an attribute macro
  --> actix-web-codegen/src/lib.rs:56:14
   |
56 | //! [TRACE]: trace
   |              ^^^^^ ambiguous link
   |
help: to link to the function, add parentheses
   |
56 | //! [TRACE]: trace()
   |              ^^^^^^^
help: to link to the attribute macro, prefix with `macro@`
   |
56 | //! [TRACE]: macro@trace
   |              ^^^^^^^^^^^

warning: `main` is both a function and an attribute macro
  --> actix-web-codegen/src/lib.rs:11:55
   |
11 | //! Used for setting up the actix async runtime. See [main] macro docs.
   |                                                       ^^^^ ambiguous link
   |
help: to link to the function, add parentheses
   |
11 | //! Used for setting up the actix async runtime. See [main()] macro docs.
   |                                                       ^^^^^^
help: to link to the attribute macro, prefix with `macro@`
   |
11 | //! Used for setting up the actix async runtime. See [macro@main] macro docs.
   |                                                       ^^^^^^^^^^

warning: `route` is a module, a function, and an attribute macro
  --> actix-web-codegen/src/lib.rs:37:32
   |
37 | //! it should respond to. See [route] macro docs.
   |                                ^^^^^ ambiguous link
   |
help: to link to the module, prefix with `mod@`
   |
37 | //! it should respond to. See [mod@route] macro docs.
   |                                ^^^^^^^^^
help: to link to the function, add parentheses
   |
37 | //! it should respond to. See [route()] macro docs.
   |                                ^^^^^^^
help: to link to the attribute macro, prefix with `macro@`
   |
37 | //! it should respond to. See [macro@route] macro docs.
   |                                ^^^^^^^^^^^

warning: 11 warnings emitted

    Checking actix-connect v2.0.0
    Checking actix-tls v2.0.0
    Checking actix-server v1.0.4
    Checking actix-testing v1.0.1
    Checking actix-http v2.2.0 (/Users/rob/Development/forks/actix-web/actix-http)
 Documenting actix-http v2.2.0 (/Users/rob/Development/forks/actix-web/actix-http)
    Checking awc v2.0.3 (/Users/rob/Development/forks/actix-web/awc)
 Documenting awc v2.0.3 (/Users/rob/Development/forks/actix-web/awc)
    Checking actix-web v3.3.2 (/Users/rob/Development/forks/actix-web)
 Documenting actix-http-test v2.1.0 (/Users/rob/Development/forks/actix-web/actix-http-test)
 Documenting actix-web v3.3.2 (/Users/rob/Development/forks/actix-web)
 Documenting actix-web-actors v3.0.0 (/Users/rob/Development/forks/actix-web/actix-web-actors)
 Documenting actix-files v0.4.1 (/Users/rob/Development/forks/actix-web/actix-files)
 Documenting actix-multipart v0.3.0 (/Users/rob/Development/forks/actix-web/actix-multipart)
warning: unresolved link to `ConnectionInfo::host`
   --> src/config.rs:144:32
    |
144 |     /// Check [ConnectionInfo](ConnectionInfo::host())
    |                                ^^^^^^^^^^^^^^^^^^^^^^ the module `config` contains no item named `ConnectionInfo`
    |
    = note: `#[warn(broken_intra_doc_links)]` on by default

warning: unresolved link to `super::dev::ConnectionInfo::realip_remote_addr`
  --> src/middleware/logger.rs:85:47
   |
85 | ///  [`ConnectionInfo::realip_remote_addr()`](super::dev::ConnectionInfo::realip_remote_addr())
   |                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the module `middleware` contains no item named `dev`

warning: unresolved link to `dev::ConnectionInfo::host`
   --> src/server.rs:216:32
    |
216 |     /// Check [ConnectionInfo](dev::ConnectionInfo::host())
    |                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^ the module `server` contains no item named `dev`

warning: 3 warnings emitted

    Finished dev [unoptimized + debuginfo] target(s) in 24.49s
     Opening /Users/rob/.cargo/target-shared/doc/actix_web/index.html
```

So I guess all the macro ones need the macro@ prefix.

**Comment by realaravinth** [comment]
I'm sorry, those warnings were confusing and multiple suggestions weren't helping either :D 

I've addressed all warnings, and ran `deadlinks` as well on this commit.

**Review by robjtede — APPROVED** [review]
awesome. thanks for tackling this.

**Pull request merged**